### ISSUE TITLE
C# Runtime Strong Name Error on Antlr4.Runtime.Standard.dll version 4.7.2

### ISSUE THREAD

**Issue by bcleech** [antlr/antlr4#2511]
I'm getting a strong name validation error on the C# runtime DLL, Antlr4.Runtime.Standard.dll, from the nuget package version 4.7.2.  I don't see this issue on 4.7.1.1.

Thanks,
Brian

>sn -v Antlr4.Runtime.Standard.dll

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Failed to verify assembly -- Strong name validation failed.

**Comment by ericvergnaud** [comment]
Do you know how to fix this?

**Comment by jlucansky** [comment]
Antlr4.Runtime.Standard 4.7.2 is quite unusable in any web application because of this :(

**Comment by ericvergnaud** [comment]
Only repeating myself: Do you know how to fix this?

**Comment by bcleech** [comment]
I'm not completely sure how to fix this.  Running sn.exe as shown above only gives the vague output that validation failed.  I'm guessing resigning the DLL in the nuget package with the private key will fix it.  I don't have access to this key, so I can't try it.

-Brian

**Comment by ericvergnaud** [comment]
I see
Maybe the private key has expired silently
I’ll check

Envoyé de mon iPhone

**Comment by juanc5ibanez** [comment]
Yep that's happening to me as well. Unfortunately it doesnt say why it fails, but it makes that version of the package unusable.  Had to go back one version to skip that error. Any ETA of a new package release?

**Comment by Andrej-Peplinski** [comment]
> Yep that's happening to me as well. Unfortunately it doesnt say why it fails, but it makes that version of the package unusable.  Had to go back one version to skip that error. Any ETA of a new package release?

I run into the same problem. Are there updates on when we can get a properly signed package?

**Comment by ericvergnaud** [comment]
A release is planned mid December 

Envoyé de mon iPhone

**Comment by Andrej-Peplinski** [comment]
> A release is planned mid December
> 
> Envoyé de mon iPhone
> …

Thanks @ericvergnaud !

I would also like to mention that I have found a (temporary) workaround that helps to avoid the assembly loading problem:
As pointed out [here ](https://forums.iis.net/t/1220602.aspx?Skipping+Strong+Name+Assembly+for+Windows+7+IIS+7+5+still+give+an+error+Couldn+t+be+Verified+)one can add `<hostingEnvironment shadowCopyBinAssemblies="false" />` in Web.config. With that I am actually able to to load Antlr4.Runtime.Standard.dll v.4.7.2 into IIS Express.

**Comment by Silvenga** [comment]
Just going mention that public signing (what Antlr is doing) isn't an option for a lot of environments:

```
You will not be able to install the assembly to the Global Assembly Cache (GAC)
You will not be able to load the assembly in an AppDomain where shadow copying is turned on.
You will not be able to load the assembly in a partially trusted AppDomain
```

Disabling Shadow copying or strong name verification aren't options for many. Shadow copying is actually needed in a lot of situations, and strong naming is needed for a lot of deployments.

**Comment by Cloudmersive** [comment]
How is this still not fixed?

**Comment by ericvergnaud** [comment]
As mentioned, a release is planned mid-december.

**Comment by ericvergnaud** [comment]
@bcleech @juanc5ibanez @Silvenga @Andrej-Peplinski @Cloudmersive hi, Terence is preparing a release, would you be able to help ensure the code signing is correct?
see #2698

**Comment by KvanTTT** [comment]
The issue can be closed since 4.7.2 is not actual for a long time.
